docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_publicly_accessible_remediation.mdx

```diff
@@ -1,6 +1,213 @@
+   - Choose the **Actions** dropdown menu, then select **Modify Cluster**.
+
+3. **Disable Public Accessibility:**
+   - In the **Cluster Properties** section, find the **Publicly Accessible** setting.
+   - Ensure the **Publicly Accessible** option is set to **No**.
+
+4. **Save Changes:**
+   - Scroll to the bottom of the page and choose **Modify Cluster** to save your changes.
+
+By following these steps, you ensure that your Redshift cluster is not publicly accessible, thereby enhancing its security.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent a Redshift Cluster from being publicly accessible using the AWS CLI, you can follow these steps:
+
+1. **Describe the Redshift Cluster to Check Current Public Accessibility:**
+   First, you need to check the current configuration of your Redshift cluster to see if it is publicly accessible.
+   ```sh
+   aws redshift describe-clusters --cluster-identifier <your-cluster-identifier>
+   ```
+
+2. **Modify the Redshift Cluster to Disable Public Accessibility:**
+   If the cluster is publicly accessible, you can modify it to disable public accessibility.
+   ```sh
+   aws redshift modify-cluster --cluster-identifier <your-cluster-identifier> --publicly-accessible false
+   ```
+
+3. **Verify the Changes:**
+   After modifying the cluster, verify that the changes have been applied correctly.
+   ```sh
+   aws redshift describe-clusters --cluster-identifier <your-cluster-identifier>
+   ```
+
+4. **Update Security Groups:**
+   Ensure that the security groups associated with your Redshift cluster do not allow inbound traffic from the public internet. You can modify the security group rules as needed.
+   ```sh
+   aws ec2 revoke-security-group-ingress --group-id <your-security-group-id> --protocol tcp --port 5439 --cidr 0.0.0.0/0
+   ```
+
+By following these steps, you can ensure that your Redshift cluster is not publicly accessible, thereby enhancing its security.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent a Redshift Cluster from being publicly accessible using Python scripts, you can use the AWS SDK for Python (Boto3). Below are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+
+3. **Modify the Redshift Cluster**:
+   Use the `modify_cluster` method to set the `PubliclyAccessible` attribute to `False`.
+
+4. **Verify the Configuration**:
+   Optionally, you can verify the configuration to ensure the cluster is not publicly accessible.
+
+Here is a Python script to prevent a Redshift Cluster from being publicly accessible:
+
+```python
+import boto3
+
+# Step 1: Initialize a Boto3 session
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+    aws_secret_access_key='YOUR_AWS_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>